### DISCUSSION TITLE
Does entt do multithreading by default?

### DISCUSSION THREAD

**Discussion by trsh** [skypjack/entt#1065] [Q&A] [answered]
Dies entt make use of multiple CPU cores?

**Comment by skypjack** [comment, marked as answer]
Well, it's like asking if a vector does mt by default. No, it does not, you can use it in a mt context but the vector on itself does nothing because it's just a container. The same is true for EnTT. It's mostly a container. It doesn't try to take over you loop, your code, your systems, your scheduling, whatever. It offers all you need to do this but it's up to you to design your code the way you like, without having to stick with a model imposed by a library.

**Reply by trsh** [reply]
Thank you for the answer. I was just looking at different ecs frameworks, like Flecs for example, and it doe's offer turning on multithreading with 1 line of code, If I am ok with Default way, it handles it.

**Reply by skypjack** [reply]
Sure thing. If you are ok with something, just go with it. I don't see the problem. 🤷‍♂️ 
EnTT was (and is) designed as a container with a bunch of side modules for mt, signalling and many other things.
However, it doesn't try to put all of them together in a pre-packed solution. That's it.
